src/main/java/io/jenkins/plugins/ml/utils/ConvertHelper.java

```diff
@@ -51,17 +52,13 @@ public static String jupyterToText(FilePath jupyterFile) throws IOException, Int
         // Convert Jupyter Notebook to JSON
         try (final InputStreamReader inputStreamReader = new InputStreamReader(jupyterFile.read(), Charset.forName("UTF-8"))) {
             Note n = new JupyterUtil().getNote(inputStreamReader, "python", "\n", "#");
-            Gson gson = new Gson();
-            JsonElement tree = gson.toJsonTree(n);
-            JsonObject obj = tree.getAsJsonObject();
-            JsonArray array = obj.get("paragraphs").getAsJsonArray();
             StringBuilder outText = new StringBuilder();
-            for (JsonElement element : array)
-                if (element.isJsonObject()) {
-                    JsonObject cell = element.getAsJsonObject();
-                    String code = cell.get("text").getAsString();
-                    outText.append(code);
-                }
+            for (Paragraph para : n.getParagraphs()) {
+                System.out.println(para.getText());
+                // skipping markdowns
+                if (para.getConfig().get("editorMode") == "ace/mode/markdown") continue;
```

Comment:
Might be useful to add a comment explaining where `"ace/mode/markdown"` came from.

Reply:
![pr_image](https://user-images.githubusercontent.com/36206296/90014484-c4b33100-dcc4-11ea-88d8-1664ebda79bb.png)
Added comment in the next commit :+1: